server/src/main/java/org/spine3/server/aggregate/AggregateRoot.java

```diff
@@ -0,0 +1,124 @@
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.spine3.server.aggregate;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.Maps;
+import com.google.protobuf.Message;
+import org.spine3.server.BoundedContext;
+
+import java.util.Map;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.spine3.server.aggregate.AggregatePartRepositoryLookup.createLookup;
+
+/**
+ * A root object for a larger aggregate.
+ *
+ * @param <I> the type for IDs of this class of aggregates
+ * @author Alexander Yevsyukov
+ */
+public class AggregateRoot<I> {
+
+    /** The map from a part class to a repository which manages corresponding aggregate part*/
+    private static final Map<Class<? extends Message>,
+                             AggregatePartRepository<?, ?>> partsAccess = Maps.newConcurrentMap();
+
+    /** The bounded context to which the aggregate belongs. */
+    private final BoundedContext boundedContext;
+
+    /** The aggregate ID. */
+    private final I id;
+
+    /**
+     * Creates an new instance.
+     *
+     * @param boundedContext the bounded context to which the aggregate belongs
+     * @param id the ID of the aggregate
+     */
+    protected AggregateRoot(BoundedContext boundedContext, I id) {
+        checkNotNull(boundedContext);
+        checkNotNull(id);
+        this.boundedContext = boundedContext;
+        this.id = checkNotNull(id);
+    }
+
+    /**
+     * Obtains the aggregate ID.
+     */
+    public I getId() {
+        return this.id;
+    }
+
+    /**
+     * Obtains the {@code BoundedContext} to which the aggregate belongs.
+     */
+    protected BoundedContext getBoundedContext() {
+        return boundedContext;
+    }
+
+    /**
+     * Obtains a part state by its class.
+     *
+     * @param partStateClass the class of the state of the part
+     * @param <S> the type of the part state
+     * @return the state of the part or a default state if the state was not found
+     */
+    protected <S extends Message, A extends AggregatePart<I, S, ?>> S getPart(Class<S> partStateClass) {
+        final AggregatePartRepository<I, A> repo = getRepository(partStateClass);
+        final AggregatePart<I, S, ?> aggregatePart = repo.loadOrCreate(getId());
+        final S partState = aggregatePart.getState();
+        return partState;
+    }
+
+    /**
+     * Obtains a repository for the passed state class.
+     *
+     * @throws IllegalStateException if a repository was not found
+     *                               or the repository does not match expectations of this {@code AggregateRoot}
+     */
+    private <S extends Message, A extends AggregatePart<I, S, ?>> AggregatePartRepository<I, A> getRepository(Class<S> stateClass) {
```

Comment:
The line looks too long. Is it possible to split it into two? Perhaps, a carriage return after `<S extends Message, ` would do.